src/os/bluestore/BlueFS.cc

```diff
@@ -166,15 +164,13 @@ void BlueFS::add_block_extent(unsigned id, uint64_t offset, uint64_t length)
   dout(1) << __func__ << " bdev " << id
           << " 0x" << std::hex << offset << "~" << length << std::dec
 	  << dendl;
-  assert(id < bdev.size());
-  assert(bdev[id]);
-  assert(bdev[id]->get_size() >= offset + length);
+  assert(id < bdev.size() && bdev[id] &&
+    bdev[id]->get_size() >= offset + length);
   block_all[id].insert(offset, length);
 
   if (id < alloc.size() && alloc[id]) {
     log_t.op_alloc_add(id, offset, length);
-    int r = _flush_and_sync_log(l);
-    assert(r == 0);
+    assert(_flush_and_sync_log(l) == 0);
```

Comment:
Please don't do this. We don't want `assert` has any side-effect, as it might be optimized to nothing(do not execute at all).

Reply:
@xiexingguo makes sense